source/material_model/melt_boukare.cc

```diff
@@ -0,0 +1,1252 @@
+
+      const double energy = 3. * number_of_atoms[i] * constants::gas_constant * Einstein_temperatures[i]
+                            * (0.5 + 1. / (std::exp(Einstein_temperatures[i] / temperature) - 1.0));
+
+      return energy;
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_molar_heat_capacity (const double temperature,
+                                   const unsigned int i) const
+    {
+      AssertThrow(temperature > 0.0,
+                  ExcMessage("The temperature has to be larger than 0!"));
+
+      const double relative_T = Einstein_temperatures[i] / temperature;
+      const double heat_capacity = 3. * number_of_atoms[i] * constants::gas_constant * std::pow(relative_T, 2)
+                                   * std::exp(relative_T) / std::pow(std::exp(relative_T) - 1.0, 2);
+
+      return heat_capacity;
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_thermal_pressure (const double temperature,
+                                const unsigned int i) const
+    {
+      const double thermal_energy = endmember_thermal_energy(temperature, i);
+      const double heat_capacity = endmember_molar_heat_capacity(reference_temperature, i);
+      const double thermal_pressure = reference_thermal_expansivities[i] * reference_bulk_moduli[i] / heat_capacity * thermal_energy;
+
+      return thermal_pressure;
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_enthalpy_thermal_addition (const double temperature,
+                                         const unsigned int i) const
+    {
+      const double addition = reference_specific_heats[i] * temperature
+                              + 0.5 * specific_heat_linear_coefficients[i] * std::pow(temperature, 2.)
+                              - specific_heat_second_coefficients[i] / temperature
+                              + 2. * specific_heat_third_coefficients[i] * sqrt(temperature)
+                              - (reference_specific_heats[i] * reference_temperature
+                                 + 0.5 * specific_heat_linear_coefficients[i] * std::pow(reference_temperature, 2.)
+                                 - specific_heat_second_coefficients[i] / reference_temperature
+                                 + 2.0 * specific_heat_third_coefficients[i] * sqrt(reference_temperature));
+
+      return addition;
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_entropy_thermal_addition (const double temperature,
+                                        const unsigned int i) const
+    {
+      const double addition = reference_specific_heats[i] * std::log(temperature)
+                              + specific_heat_linear_coefficients[i] * temperature
+                              - 0.5 * specific_heat_second_coefficients[i] / std::pow(temperature, 2.)
+                              - 2.0 * specific_heat_third_coefficients[i] / sqrt(temperature)
```

Comment:
please check all occurences of `sqrt` in the file if they need a change
```suggestion
                              - 2.0 * specific_heat_third_coefficients[i] / std::sqrt(temperature)
```